config/params/config.go

```diff
@@ -400,41 +406,38 @@ func (b *BeaconChainConfig) TargetBlobsPerBlock(slot primitives.Slot) int {
 	return b.DeprecatedMaxBlobsPerBlock / 2
 }
 
-// MaxBlobsPerBlock returns the maximum number of blobs per block for the given slot.
 func (b *BeaconChainConfig) MaxBlobsPerBlock(slot primitives.Slot) int {
 	epoch := primitives.Epoch(slot.DivSlot(b.SlotsPerEpoch))
 
-	if epoch >= b.FuluForkEpoch {
-		return b.DeprecatedMaxBlobsPerBlockFulu
+	if len(b.BlobSchedule) > 0 {
+		// Assume BlobSchedule is already sorted ascending by Epoch
```

Comment:
I think this assumption is bad when we could implement a struct or something to reliably return the correct data, sorted or unsorted.

Reply:
But that adds implementation complexity and comes at a cost.

My thinking is there's no real difference between using a config with completely wrong values and using one where the values are just in the wrong order. Either way, your code won’t work. If the blob schedule isn't sorted correctly, it’s just as bad as having invalid values, and the spec does require it to be sorted.